python/tests/transformers/keras_transformer_test.py

```diff
@@ -0,0 +1,145 @@
+        return glorot_uniform(seed=self.RANDOM_SEED)
+
+    def getKerasModel(self):
+        """
+        Build and return keras model for (binary) classification on one-dimensional data.
+
+        The model has a single hidden layer that feeds into a single-unit, sigmoid-activated output
+        unit. Weights of the hidden & output layer are drawn from a zero-mean uniform distribution.
+        """
+        model = Sequential()
+
+        # We add a vanilla hidden layer, specifying the initializer (RNG) for the kernel & bias
+        # weights
+        model.add(Dense(units=50, input_shape=(self.NUM_FEATURES, ),
+                        bias_initializer=self._getKerasModelWeightInitializer(),
+                        kernel_initializer=self._getKerasModelWeightInitializer()))
+        model.add(Activation('relu'))
+
+        # We project onto a single unit output layer, and squash it with a sigmoid:
+        model.add(Dense(units=1, bias_initializer=self._getKerasModelWeightInitializer(),
+                        kernel_initializer=self._getKerasModelWeightInitializer()))
+        model.add(Activation('sigmoid'))
+        return model
+
+    def prepareKerasModelFile(self, model, filename):
+        """
+        Saves the passed-in keras model to a temporary directory with the specified filename.
+        """
+        path = os.path.join(self.tempdir, filename)
+        model.save(path)
+        return path
+
+    def executeKerasModel(self, df, model, input_col, id_col):
+        """
+        Given an input DataFrame, applies the passed-in Keras model to the data in the specified
+        input column, returning a numpy array of the model output sorted by increasing row ID (where
+        row ID is contained in id_col).
+        """
+        # Collect dataframe, sort rows by ID column
+        rows = df.select(input_col, id_col).collect()
+        rows.sort(key=lambda row: row[id_col])
+        # Get numpy array (num_examples, num_features) containing input data
+        x_predict = np.array([row[input_col] for row in rows])
+        return model.predict(x_predict)
+
+    def transformOutputToComparables(self, final_df, id_col, output_col):
+        """
+        Given the output of KerasTransformer.transform(), collects transformer output and
+        returns a list of model predictions ordered by row id.
+
+        params:
+        :param final_df: DataFrame, output of KerasTransformer.transform()
+        :param id_col: String, Column assumed to contain a row ID (Long)
+        :param output_col: String, Column containing transform() output. We assume that for
+                           each row, transform() has produced a single-element array containing a
+                           Keras model's prediction on that row)
+        """
+        collected = final_df.collect()
+        collected.sort(key=lambda row: row[id_col])
+        return [row[output_col][0] for row in collected]
+
+    def test_keras_transformer_single_dim(self):
```

Comment:
I'd put the test_ methods at the top of the class since they are what matter the most